src/API.jl

```diff
@@ -18,16 +18,23 @@ preview_info() = printstyled("───── Preview mode ─────\n"; c
 
 include("generate.jl")
 
-function check_package_name(x::AbstractString)
+function check_package_name(x::AbstractString, mode::Symbol)
     if !(occursin(Pkg.REPLMode.name_re, x))
-         pkgerror("$x is not a valid packagename")
+        islikely_urlorpath(x) = any(occursin.(['\\','/'], x))
+        message = "$x is not a valid packagename."
+        if islikely_urlorpath(x)
+            message *= " $x appears to be a URL or path. Did you mean:\n    Pkg.$mode(PackageSpec([url|path]=\"$x\"))?"
+        end
+        pkgerror(message)
     end
-    return PackageSpec(x)
+    return PackageSpec(; name = x)
 end
 
 add_or_develop(pkg::Union{AbstractString, PackageSpec}; kwargs...) = add_or_develop([pkg]; kwargs...)
-add_or_develop(pkgs::Vector{<:AbstractString}; kwargs...) =
-    add_or_develop([check_package_name(pkg) for pkg in pkgs]; kwargs...)
+add_or_develop(pkgs::Vector{<:AbstractString}; mode::Symbol, kwargs...) =
+    add_or_develop([check_package_name(pkg, mode) for pkg in pkgs]; kwargs...)
+add_or_develop(; url = nothing, path = nothing, kwargs...) =
```

Comment:
A few concerns regarding this method:
- The source location is initially specified only at `(::PackageSpec).repo.url`, not in `path`. If you look at `API.Package`, you will see that both `url` and `path` get mapped to the same field internally. This is a bit confusing because users expect `path` to mean local filesystem path, but `path` has a different meaning internally.
- It does not make sense to specify both a `url` and a `path` for a single package. `API.Package` guards against this, but this function would circumvent that input checking.

Reply:
Oops, that was a leftover, since we agreed to just do PackageSpec(). I had checked that passing `nothing` as default to both url and path to PackageSpec allowed a user to specify just one and it would fail if both were there, but could have missed something. Anyway, I just removed this function version.